src/common/validate.cpp

```diff
@@ -40,5 +43,39 @@ wxValidator::~wxValidator()
 {
 }
 
+void wxValidator::DoProcessEvent(wxValidationEvent& event)
+{
+    // First, give a chance to the m_validatorWindow to process the event
+    // by itself. If no handler was found to process the event, or it was
+    // just skipped, process it here by popping up the error message using
+    // a wxRichToolTip or wxMessageBox (depending on the configuration)
+
+    if ( !m_validatorWindow->GetEventHandler()->ProcessEvent(event) )
+    {
+    #if wxUSE_RICHTOOLTIP
+        wxRichToolTip tip(_("Validation conflict"), event.GetErrorMessage());
```

Comment:
I think this is an incompatible change as the existing code using non-text validators would start showing tooltips (not too bad but can still be confusing, especially if the program does something similar on its own) or message boxes (really bad) by default.

I'm not sure what the solution is though...

Reply:
So we should continue doing as we've already done and let users decide how to popup validation errors on their own.